### ISSUE TITLE
Incorrect IDE0004 (Cast is redundant) in pattern matching

### ISSUE THREAD

**Issue by sharwell** [dotnet/roslyn#19987]
**Version Used**: 15.3 Preview 1

🔗 Originally reported by [Thomas Castiglione](https://developercommunity.visualstudio.com/users/21348/873dad1a-b02f-4f5a-8da9-37ef170c009b.html).

**Steps to Reproduce**:

```csharp
using System;

namespace ConsoleApp4
{
    class Program
    {
        static void Main(string[] args)
        {
            object x = 1f;
            switch (x)
            {
            case (float)1:
                Console.WriteLine("one");
                break;

            default:
                Console.WriteLine("not one");
                break;
            }
        }
    }
}
```

**Expected Behavior**:

* IDE0004 is not reported on the cast in `(float)1`
* Fixing all violations of IDE0004 does not remove the cast

**Actual Behavior**:

* The cast is removed when fixing IDE0004, resulting in a change in program behavior
